### ISSUE TITLE
LINUX-SETUP-SHORTGUIDE.md

### ISSUE THREAD

**Issue by derhier** [RainbowMiner/RainbowMiner#2602]
"chmod +x \*.sh" should be "chmod +x *.sh"

**Comment by slappy6212** [comment]
Yes i waste a couples of minutes on updating the guide for myself....for future usage. My O/C are still not working...i am troobleshooting this actually.

cd RainbowMiner
chmod +x *.sh
sudo ./install.sh

sudo apt update
sudo apt install samba

sudo nano /etc/samba/smb.conf

Add this to the file

[RainbowMiner_Machine]
path = /home
writeable = yes
browseable = yes
public = yes
create mask = 0777
directory mask = 0777
force user = root

[RainbowMiner_Folder]
path = /home/user/RainbowMiner
writeable = yes
browseable = yes
public = yes
create mask = 0777
directory mask = 0777
force user = root

sudo service smbd restart

smbpasswd -a root
chosse a password.

Now when connecting from another machine, the user is root and the password is the one previously created.

Modify Rainbow miner config.txt

  "EnableOCLinuxSetAllPStates": "0",
  "EnableOCLinuxForcePState": "0",
  "EnableLinuxHeadless": "1",
  "LinuxDisplay": "0",
  "LinuxXAuthority": "/usr/lib/xorg",
  "EnableMinersAsRoot": "0",


sudo nano /etc/rc.local
add the following to rc.local

sleep 15
/home/user/RainbowMiner/start-screen.sh
exit 0

**Comment by timvis** [comment]
maybe you want to add this too. If wanted, it also allows to run RBM with root permissions.

### Use systemd to start RainbowMiner as a service and control it
**As root user create a file with a text editor, '/etc/systemd/system/rainbowminer.service' with the following content**
(also adapt the username and path to RainbowMiner)

```
[Unit]
Description=starts RainbowMiner as a service
Wants=network.target

[Service]
Type=simple

User=USERNAME  ## your username here, or just comment it out for root permissions
ExecStart=/YOURPATH/RainbowMiner/start-nohup.sh

KillMode=none
TimeoutStopSec=15
ExecStop=/YOURPATH/RainbowMiner/stopp.sh

[Install]
WantedBy=multi-user.target
```

Then do 'systemctl daemon-reload' once, so that the launchsystem get aware of the added service.
Do 'systemctl _enable_ rainbowminer.service' once to enable automatic start at boot time.

Do 'systemctl _stop_ rainbowminer.service' for a clean shutdown anytime.
Do 'systemctl _start_ rainbowminer.service' to bring it up again.

**Comment by timvis** [comment]
The 'killmode=none' and 'timeout=15s' is a workaround. I didn't find a better way. If someone knows I am glad to learn how it's done.

**Comment by slappy3** [comment]
After RB is started, how to auto start /home/user/RainbowMiner/minerlog.sh ? Having  the miners in actions on the screen.

**Comment by timvis** [comment]
Adding it to '/home/yourmininguser/.bash_profile' works for me in Ubuntu 22.04 LTS. Then it is shown directly after the login with 'yourmininguser' . I tested it over ssh.

**Comment by slappy3** [comment]
No .bash_profile  found in user folder so i created one but no luck here is the content /home/user/RainbowMiner/minerlog.sh

**Comment by timvis** [comment]
It depends on your Linux-distro how it is done. Maybe a file '.bash_login' exists in the userfolder? the command 'ls -a' shows all files, also the hidden ones starting with a dot. For any user the scripts in '/etc/profile.d/' will be started.

**Comment by slappy3** [comment]
I found the problem. I need to modify the minerlog.sh file after placing it to a new folder. It need an absolute path to work with this situation. Or maybe we can create a shortcut like we do in windows but i dont know how.

**Comment by timvis** [comment]
SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"

or 

cd "$(dirname "$0")"
**SCRIPT_DIR="$(pwd)"**
rbmtail $SCRIPT_DIR/Logs miner

**Comment by timvis** [comment]
Still if you need windows-type shortcuts, they are called links. A symbolic link (SYMLINK) is usually the sufficient and flexible type. They are created with 'ln -s TARGETFILE LINKNAME'
